src/services/miiverse-api/routes/communities.js

```diff
@@ -87,4 +88,34 @@ router.get('/:appID/posts', async function (req, res) {
     res.send(await comPostGen.PostsResponse(posts, community, options));
 });
 
+// Handler for POST on '/v1/communities'
+router.post('/', multer().none(), async function (req, res) {
+    const paramPack = util.decodeParamPack(req.headers["x-nintendo-parampack"]);
+    let parent_community = await database.getCommunityByTitleIDs(paramPack.title_id);
+    if(!parent_community) res.sendStatus(404);
+
+    let num_communities = await COMMUNITY.count();
+    let new_community = new COMMUNITY({
+        platform_id: 0, // WiiU
+        name: req.body.name,
+        description: req.body.description,
+        open: true,
+        allows_comments: true,
+        type: 1,
+        parent: parent_community.community_id,
+        admins: parent_community.admins,
+        icon: req.body.icon,
+        title_id: paramPack.title_id,
+        community_id: (parseInt(parent_community.community_id) + (5000 * num_communities)).toString(),
+        olive_community_id: (parseInt(parent_community.community_id) + (5000 * num_communities)).toString(),
```

Comment:
Is there a reason that these IDs are generated this way? Or just a random choice. This isn't very safe, since it's possible for 2 requests to hit this endpoint at the same time and have `num_communities` be the same value, resulting in a duplicate ID

Reply:
The reason is i have no clue how the community IDs are generated, so we will need to work on a global solution. Our current solution is creating the IDs manually for each community. 

Since this endpoint creates a subcommunity programmatically, we need a better solution to be implemented.